src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.html

```diff
@@ -36,11 +40,46 @@ <h4>{{ 'artemisApp.courseStatistics.courseAverage' | artemisTranslate }}: {{ cou
         </ngx-charts-bar-vertical>
     </div>
     <fa-icon
-        *ngIf="exerciseAverageScores.length > 10 && exerciseAverageScores[10 + currentPeriod] != null"
+        *ngIf="currentSize > 10 + currentPeriod; else placeholder"
         [icon]="faArrowRight"
         size="3x"
-        class="col-md-1 d-flex justify-content-start align-items-center"
+        class="col d-flex align-items-center switch-forward"
         role="button"
         (click)="switchTimeSpan(RIGHT)"
     ></fa-icon>
+    <ng-template #placeholder>
+        <div class="switch-forward-placeholder"></div>
+    </ng-template>
+    <div class="col d-flex flex-column align-items-center justify-content-center ms-4">
+        <div *ngIf="typeSet.size > 0" class="filter mb-3" aria-label="Filter Dropdown" ngbDropdown>
+            <button
+                class="btn"
+                [ngClass]="{ 'btn-secondary': numberOfActiveFilters === 0, 'btn-success': numberOfActiveFilters > 0 }"
+                ngbDropdownToggle
+                id="filter-dropdown-button"
+            >
+                <fa-icon [icon]="faFilter"></fa-icon>
+                <span>{{ 'artemisApp.courseOverview.exerciseList.filter' | artemisTranslate: { num: numberOfActiveFilters } }}</span>
+            </button>
+            <ul ngbDropdownMenu class="checkbox-menu text-nowrap pe-2" aria-labelledby="filter-dropdown-button">
+                <li>
+                    <b class="ms-2">{{ 'artemisApp.exercise-scores-chart.includeType' | artemisTranslate }}</b>
+                </li>
+                <li *ngFor="let type of typeSet">
+                    <label>
+                        <input class="ms-2" (change)="toggleType(type)" [checked]="chartFilter.get(type.toLowerCase().replace('_', '-'))" type="checkbox" />
```

Comment:
This refers to my previous comment: You could reuse a new ``getChartFilter`` method here as well